File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    # 여벌 체육복이 있는 학생 -> 체육복 빌려줄 수 있음
    # 학생들 번호: 체격순
    # 앞 번호, 뒷 번호 학생에게만 빌려줄 수 있음
    # 최대한 많은 학생이 체육 수업을 듣도록(여벌 + 도난당했지만 채육복 빌린 학생)
    
    # 학생 만들기
    st = []
    for i in range(1, n+1):
        st.append(i)
    # 체육복 빌릴 수 있는 후보 만들기
    # 여분 가져온 학생이 도난 당했을 경우 제외
    lost, reserve = list(set(lost)-set(reserve)), list(set(reserve)-set(lost))
    
    ca = []
    for l in lost:
        ca.append((l-1, l+1))
    
    for r in reserve:
        for c in ca:
            if r in c: # 체육복 빌릴 수 있는 경우
                answer+=1
                ca.remove(c) # 체육복 후보에서 제외
                break

    # 학생 - 도난 당한 학생 + 빌릴 수 있는 학생
    return n - len(lost) + answer
                
    
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        // 전체 학생수: n, 체육복 도난당한 학생들 번호: lost, 여벌 체육복 학생들 번호: reserve
        // 체육수업을 들을 수 있는 학생 최댓값?
        
        int answer = 0;
        
        // 전체 학생 배열
        int[] student = new int[n];
        // 모든 학생이 체육복을 1개씩 가지고 있다고 초기화
        Arrays.fill(student,1);
        // 체육복을 도난당한 학생 처리
        for(int l:lost){
            student[l-1]--; // 인덱스화
        }
        // 여벌체육복이 있을 경우 체육복 +1
        for(int r:reserve){
            student[r-1]++;
        }
        // 체육복 빌려주기
        for(int i=0; i<n; i++){
            if(student[i]==0) {// 도난 당했을경우
                if(i>0 && student[i-1]==2) {// 내 앞 학생이 체육복 2개일 경우
                    student[i]++;
                    student[i-1]--;
                }else if(i<n-1 && student[i+1]==2){ // 마지막 인덱스 x && 내 뒤 학생이 체육복 2개일 경우
                    student[i]++;
                    student[i+1]--;
                }
            }
                
        }
        // 체육수업을 들을 수 있는 학생 count
        for(int i=0;i<n;i++){
            if(student[i]>=1){
                answer++;
            }
        }

        
        
        return answer;
    }
}